src/main/java/org/mskcc/limsrest/controller/CmoRequestMetaDbPublisher.java

```diff
@@ -0,0 +1,108 @@
+package org.mskcc.limsrest.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.shared.SampleManifest;
+import org.mskcc.cmo.shared.neo4j.CmoRequestEntity;
+import org.mskcc.limsrest.ConnectionLIMS;
+import org.mskcc.limsrest.service.GetRequestSamplesTask;
+import org.mskcc.limsrest.service.GetSampleManifestTask;
+import org.mskcc.limsrest.service.SampleManifestExtended;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import static org.mskcc.limsrest.util.Utils.getResponseEntity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.web.server.ResponseStatusException;
+
+/**
+ * Publishes CMO requests as package to MetaDB with NATS messaging.
+ * Published package includes request ID and all samples for request.
+ * @author ochoaa
+ */
+@ComponentScan("org.mskcc.cmo.messaging")
+@RestController
+@RequestMapping("/")
+public class CmoRequestMetaDbPublisher {
+
+    @Autowired
+    private Gateway messagingGateway;
+
+    private static Log log = LogFactory.getLog(CmoRequestMetaDbPublisher.class);
+    private final ConnectionLIMS conn;
+
+    public CmoRequestMetaDbPublisher(ConnectionLIMS conn) throws Exception {
+        this.conn = conn;
+        if (messagingGateway == null) {
+            log.error("Error establishing connection to NATS server, messagingGateway is NULL");
+        } else {
+            if (!messagingGateway.isConnected()) {
+                messagingGateway.connect();
+            }
+        }
+    }
+
+    @GetMapping("/publishCmoRequestToMetaDb")
+    public boolean getContent(@RequestParam(value = "request") String requestId,
+                                                          HttpServletRequest request) {
+        log.info("/publishCmoRequestToMetaDb for request: " + requestId + " " + request.getRemoteAddr());
+
+        if (!Whitelists.requestMatches(requestId)) {
```

Comment:
@l-wagner decided to use the same logic here that is used elsewhere - do you think we should have this check here as well or is it not necessary?

Reply:
It definitely doesn't hurt having it.